advisor_engine/endpoints.py

```diff
@@ -33,12 +34,19 @@ def handle_system_get(insights_id: UUID = '00000000-0000-0000-0000-000000000000'
 def handle_system_exists(insights_id: UUID = '00000000-0000-0000-0000-000000000000'):
     return {"id": insights_id}
 
-async def handle_insights_archive(file: Optional[UploadFile] = File(None),
+async def handle_insights_archive(request: Request,
+                                  file: Optional[UploadFile] = File(None),
                                   test: str=Form(None)):
     # insights-client --test-connection
     # Just want to send back a 200 for the Client/Satellite
     if test: return Response(status_code=status.HTTP_200_OK)
-    else:        
+    else:
+        content_length = int(request.headers.get('content-length'))
```

Comment:
Non-integer values will cause HTTP 500, instead of 400.

Reply:
@vkrizan I could not replicate this. How were you able to see this so I may replicate?

Reply:
fast api validates that header before it reaches the handler.